### DISCUSSION TITLE
Why not create a fieldPool like cePool(CheckEntry) to reuse Fields

### DISCUSSION THREAD

**Discussion by liang09255** [uber-go/zap#1291] [Q&A]
I see that when creating each field, a new Field object is created. 
Why not create a fieldPool like cePool(CheckEntry) to reuse Fields

**Comment by prashantv** [comment]
Fields are used as value types passed on the stack, so there's no need for pooling.

These (old) PRs have some context for how we landed on a stack-allocated union vs pooling a heap-allocated field:
https://github.com/uber-go/zap/pull/10
https://github.com/uber-go/zap/pull/13
